cloud-controller-manager/do/kubernetes_lb_validator.go

```diff
@@ -0,0 +1,185 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package do
+
+import (
+	"context"
+	"fmt"
+	"github.com/digitalocean/godo"
+	"github.com/go-logr/logr"
+	"golang.org/x/oauth2"
+	v1 "k8s.io/api/core/v1"
+	"net/http"
+	"os"
+	"strconv"
+
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// LBService represents the v1.service lb object data
+type LBService struct {
+	APIVersion string
+	Kind       string
+	Metadata   Metadata
+	Spec       map[string]interface{}
+	Selector   map[string]interface{}
+	Type       string
+}
+
+// Metadata represents the metadata field in the lb service object
+type Metadata struct {
+	Annotations interface{} `json:"annotation data,omitempty"`
+	Name        string
+	Namespace   string
+}
+
+// DOKSLBServiceValidator validates service type LB
+type KubernetesLBServiceValidator struct {
+	decoder *admission.Decoder
+	Log     logr.Logger
+	gClient *godo.Client
+}
+
+func initDOClient() (doClient *godo.Client, err error) {
+	token := os.Getenv(doAccessTokenEnv)
+
+	if token == "" {
+		return nil, fmt.Errorf("environment variable %q is required", doAccessTokenEnv)
+	}
+
+	tokenSource := &tokenSource{
+		AccessToken: token,
+	}
+
+	var opts []godo.ClientOpt
+
+	opts = append(opts, godo.SetUserAgent("digitalocean-webhook-server/"+version))
+
+	oauthClient := oauth2.NewClient(oauth2.NoContext, tokenSource)
+	doClient, err = godo.New(oauthClient, opts...)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create godo client: %s", err)
+	}
+
+	return doClient, nil
+}
+
+// DOKSLBServiceValidator ...
+func (v *KubernetesLBServiceValidator) Handle(ctx context.Context, req admission.Request) admission.Response {
+	svc := &v1.Service{}
+
+	v.Log.V(6).Info("decoding received request")
+	err := v.decoder.Decode(req, svc)
+	if err != nil {
+		v.Log.Error(err, "failed to decode request")
+		fmt.Errorf("failed to decode request: %s", err)
+		return admission.Denied("failed to decode request")
+	}
+
+	v.Log.V(6).Info("checking received request")
+	if svc.Spec.Type != v1.ServiceTypeLoadBalancer || svc.DeletionTimestamp != nil {
+		return admission.Allowed("the service object is either not a load balancer or being deleted")
+	}
+
+	// initialize DO Client if not already initialized
+	if v.gClient == nil {
+		doClient, err := initDOClient()
+		if err != nil {
+			v.Log.Error(err, "failed to initialize DO client")
+			return admission.Errored(http.StatusConflict, err)
+		}
+		v.gClient = doClient
+	}
+
+	clusterID := os.Getenv(doClusterIDEnv)
+	if clusterID == "" {
+		fmt.Println("missing cluster id")
+	}
+	region := "nyc1"
+	if region == "" {
+		fmt.Println("missing region id")
+	}
+
+	nodePools, _, err := v.gClient.Kubernetes.ListNodePools(ctx, clusterID, &godo.ListOptions{})
```

Comment:
After speaking a bit about this PR and it's use of the DOKS client w/ Timo and Varsha. They had some good points as to why we shouldn't use it and how we could address it.

Let's consider making LB create requests w/ tags and not Droplet IDs (https://docs.digitalocean.com/reference/api/kafka-beta-api-reference/#operation/loadBalancers_create)

That way... we could just use droplet tags (not droplet IDs) to make that request since droplet tags could simply be the 3 we use for droplets associated w/ a cluster:
- `k8s`
- `k8s:<cluster-id>`
- `k8s:worker`

In other words, you can construct these tags in-memory in your code, don't need to pull them from anywhere since we already have in-memory access to the cluster ID.